### PR TITLE
`completion-tool list-executables` should list executable names

### LINKED ISSUE #8033: `swift package completion-tool list-executables` lists names of targets instead of names of executables
### Is it reproducible with SwiftPM command-line tools: `swift build`, `swift test`, `swift package` etc?

- [X] Confirmed reproduction steps with SwiftPM CLI. The description text _must_ include reproduction steps with either of command-line SwiftPM commands, `swift build`, `swift test`, `swift package` etc.

### Description

`swift package completion-tool list-executables` should list names of executables but in reality it lists names of executable targets

### Expected behavior

As mode `list-executables` suggests, I expect `swift package completion-tool list-executables` list names of executables

### Actual behavior

`swift package completion-tool list-executables` lists names of executable targets

### Steps to reproduce

Run `swift package completion-tool list-executables` on any swift package having executable targets where name of executable doesn't match name of target.

### Swift Package Manager version/commit hash

Swift Package Manager - Swift 6.0.0-dev

### Swift & OS version (output of `swift --version ; uname -a`)

```
swift --version ; uname -a 
swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
Darwin VV-LIN-MBP.local 24.0.0 Darwin Kernel Version 24.0.0: Tue Sep 24 23:36:26 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T8103 arm64
```

### PR DESCRIPTION
**Pull request by plemarquand**
Update the completion tool so it prints the list of executable names instead of target names. The executable products names determines the name fo the actual binary written to disk, which is what you'd want from a completion-tool.

That patch also adds a few tests for completion-tool around this specific behaviour. There weren't existing tests for the completion-tool. More tests to follow in a subsequent PR focusing on testing this area specifically.

Fixes #8033

### CODE REVIEW

File: Fixtures/Miscellaneous/DifferentProductTargetName/main.swift [hunk 1]
```diff
@@ -0,0 +1,3 @@
+public func foo() {
+	{}()
+}
```
Comment (MaxDesiatov) [inline Fixtures/Miscellaneous/DifferentProductTargetName/main.swift:3]: This is not called from anywhere, so the file could be empty instead?

File: Fixtures/Miscellaneous/DifferentProductTargetName/main.swift [hunk 2]
```diff
@@ -0,0 +1,2 @@
+public func foo() {
+}
```
[the remark this answers is not in the document]
Comment (plemarquand) [reply Fixtures/Miscellaneous/DifferentProductTargetName/main.swift]: 🥴 done

**Review by MaxDesiatov — APPROVED** [review]

File: Fixtures/Miscellaneous/DifferentProductTargetName/Package.swift [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// swift-tools-version:5.2
+import PackageDescription
+
+let package = Package(
+    name: "Foo",
+    products: [
+        .executable(name: "Foo", targets: ["Foo"]),
```
Comment (MaxDesiatov) [inline Fixtures/Miscellaneous/DifferentProductTargetName/Package.swift:7]: Name of the product and the target are the same, why is the fixture called `DifferentProductTargetName` then?

Comment (plemarquand) [reply Fixtures/Miscellaneous/DifferentProductTargetName/Package.swift]: Good catch

**Comment by plemarquand** [comment]
@swift-ci test

**Pull request merged**